docs/reference/index.rst

```diff
@@ -14,6 +14,7 @@ Reference Documentation
    generators
    sampler_composites/index
    sampleset
+   typing
```

Comment:
These are not alphabetical, so why not put this after Exceptions? I think users will likely first need utilities and the pages related to variables and their types before they get to this. 
I can live with it here though too.

Reply:
Sure, can move it.